### PR TITLE
BackendSrv: Fixes POST body for form data

### LINKED ISSUE #21688: Prometheus POST queries to grafana backend are truncated
**What happened**:
We updated grafana to 6.7.0-e861b821pre

our dashboards don't work at all:

allerr

**What you expected to happen**:
It works

**How to reproduce it (as minimally and precisely as possible)**:

take grafana master

add a valid prometheus backend with POST method

click save

**Environment**:
- Grafana version: 6.7.0-e861b821pre from the rpm
- Data source type & version: prometheus

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Fixes the body for Form data

**Which issue(s) this PR fixes**:
Fixes #21688

**Special notes for your reviewer**:
Related to #21639

### CODE REVIEW

**Review by kaydelaney — APPROVED** [review]
LGTM

**Comment by hugohaggmark** [comment]
> LGTM

Did you have a chance to test this with Elastic @kaydelaney

**Review by dprokop — APPROVED** [review]
LGTM!

**Comment by hugohaggmark** [comment]
@alexanderzobnin can you check that this change doesn't break the Splunk plugin?

**Review by mckn — APPROVED** [review]
LGTM

File: public/app/core/specs/backend_srv.test.ts [hunk 1]
```diff
@@ -155,13 +155,17 @@ describe('backendSrv', () => {
...
+      ${'GET'}     | ${{ Auth: 'Some Auth', 'Content-Type': 'application/x-www-form-urlencoded' }} | ${{ data: 'Some data' }}           | ${{ method: 'GET', headers: { 'Content-Type': 'application/x-www-form-urlencoded', Accept: 'application/json, text/plain, */*', Auth: 'Some Auth' }, body: new URLSearchParams({ data: 'Some data' }) }}
+      ${'GET'}     | ${{ Auth: 'Some Auth', 'Content-Type': 'application/x-www-form-urlencoded' }} | ${'some data'}                     | ${{ method: 'GET', headers: { 'Content-Type': 'application/x-www-form-urlencoded', Accept: 'application/json, text/plain, */*', Auth: 'Some Auth' }, body: 'some data' }}
+      ${'GET'}     | ${{ Auth: 'Some Auth', 'Content-Type': 'application/x-www-form-urlencoded' }} | ${'{"data":{"test":"Some data"}}'} | ${{ method: 'GET', headers: { 'Content-Type': 'application/x-www-form-urlencoded', Accept: 'application/json, text/plain, */*', Auth: 'Some Auth' }, body: '{"data":{"test":"Some data"}}' }}
```
Comment (roidelapluie) [inline public/app/core/specs/backend_srv.test.ts]: should we have some POST test?

Comment (hugohaggmark) [reply public/app/core/specs/backend_srv.test.ts]: That would be just for the sake of examples, we actually don't test posting or getting in this test. But I can see your point.

**Pull request merged**